marshal/consumer.go

```diff
@@ -639,8 +639,9 @@ func (c *Consumer) terminateAndCleanup(release bool, remove bool) bool {
 
 	latestTopicClaims := make(map[string]bool)
 	releasedTopics := make(map[string]bool)
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+
```

Comment:
but, due to check above, atomic.CompareAndSwapInt32(c.alive, 1, 0), only one will be able to execute this code. I'm not sure how this would fix the race.